plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -454,4 +429,37 @@ public static List<ParquetReaderColumn> createParquetReaderColumns(List<HiveColu
                 })
                 .collect(toImmutableList());
     }
+
+    private static Optional<Type> getParquetColumnType(HiveColumnHandle column, MessageType messageType, boolean useParquetColumnNames)
+    {
+        if (useParquetColumnNames) {
+            return Optional.ofNullable(getParquetTypeByName(column.getBaseColumnName(), messageType));
+        }
+        if (column.getBaseHiveColumnIndex() < messageType.getFieldCount()) {
+            return Optional.of(messageType.getType(column.getBaseHiveColumnIndex()));
+        }
+
+        return Optional.empty();
+    }
+
+    private static List<Type> dereferenceSubFieldTypes(GroupType baseType, HiveColumnHandle column)
+    {
+        if (baseType == null || column.getHiveColumnProjectionInfo().isEmpty()) {
```

Comment:
is baseType null possible in any codepath ? If not, I'd put a checkArgument instead

Reply:
this cannot be null, since it were called in two places within the class, and both calls have checks to check if the Optional<Type> columnType is indeed empty.